lld/include/lld/Common/BPSectionOrdererBase.inc

```diff
@@ -98,10 +102,11 @@ static SmallVector<std::pair<unsigned, UtilityNodes>> getUnsForCompression(
     // Merge sections that are nearly identical
     SmallVector<std::pair<unsigned, SmallVector<uint64_t>>> newSectionHashes;
     DenseMap<uint64_t, unsigned> wholeHashToSectionIdx;
+    unsigned threshold = sectionHashes.size() > 10000 ? 5 : 0;
```

Comment:
@ellishg With the existing threshold 5, wholeHash is almost always 0 for smaller applications. I guess 5 was picked to improve the performance of  larger applications.

Reply:
Yeah, 5 was picked because we saw good results for large binaries. If we wanted to generalize this we could compute something like the P90 of the hash frequency, which would be the set of hashes that are repeated more than 90% of the hashes. But I would like to experiment with this.

Reply:
I think it's important for tests to show that `--bp-compression-sort=data` actually does something. Therefore, the threshold has to be lowered so that data sections will not all be considered duplicates.... 

I think `sectionHashes.size() > 10000` will work for large applications like yours.

(In the test, --bp-compression-sort=both and --bp-compression-sort=data exhibit the same behavior. Ideally 'both' should actually do something with functions. Reserved for a future patch.)

Reply:
I built some large binaries with this change and I didn't see any large differences, so I think this is ok. I also realized that there is a bug. If `wholeHash` is zero, we should not insert into `duplicateSectionIdxs` because they have no hashes in common.

The more I think about this the more I realize this might be able to be improved quite a bit. I can imagine a set of small outlined functions that differ in one instructions. Ideally these should be in `duplicateSectionIdxs`, but this code won't catch that case.